doc/ss_logger.rst

```diff
@@ -0,0 +1,234 @@
+Once initialized, a user may use the instance to log a message using one of the four
+logging functions.
+
+To use the SmartSim logger within a Python script, import the required `get_logger`
+function from the `log` module:
+
+.. code-block:: python
+
+      from smartsim.log import get_logger
+
+Next, initialize an instance of the logger and provide a logger `name`:
+
+.. code-block:: python
+
+      logger = get_logger("SmartSim")
+
+To demonstrate full functionality of the SmartSim logger, we include all log
+functions in the Python driver script with log messages:
+
+.. code-block:: python
+
+      logger.info("This is a message")
+      logger.debug("This is a debug message")
+      logger.error("This is an error message")
+      logger.warning("This is a warning message")
+
+Execute the script *without* setting the `SMARTSIM_LOG_LEVEL`. Remember that `SMARTSIM_LOG_LEVEL`
+defaults to `info`. When we execute the script, the following messages will print to stdout:
+
+.. code-block:: stdout
+
+    11:15:00 system.host.com SmartSim[130033] INFO This is a message
+    11:15:00 system.host.com SmartSim[130033] ERROR This is an error message
+    11:15:00 system.host.com SmartSim[130033] WARNING This is a warning message
+
+Notice that the `debug` function message was filtered. This is because by using
+a lower verbosity level (`info`), we instruct SmartSim to omit the higher verbosity level messages (`debug` and `developer`).
+
+Next, set `SMARTSIM_LOG_LEVEL` to `debug`:
+
+.. code-block:: bash
+
+    export SMARTSIM_LOG_LEVEL=debug
+
+When we execute the script again,
+the following messages will print to stdout:
+
+.. code-block:: stdout
+
+    11:15:00 system.host.com SmartSim[65385] INFO This is a message
+    11:15:00 system.host.com SmartSim[65385] DEBUG This is a debug message
+    11:15:00 system.host.com SmartSim[65385] ERROR This is an error message
+    11:15:00 system.host.com SmartSim[65385] WARNING This is a warning message
+
+Notice that all log messages print to stdout. By using a higher verbosity level (`debug`),
+we instruct SmartSim to print all log functions at and above the level.
+
+Next, set `SMARTSIM_LOG_LEVEL` to `developer`:
+
+.. code-block:: bash
+
+    export SMARTSIM_LOG_LEVEL=developer
+
+When we execute the script,
+the following messages will print to stdout:
+
+.. code-block:: stdout
+
+    11:15:00 system.host.com SmartSim[65385] INFO This is a message
+    11:15:00 system.host.com SmartSim[65385] DEBUG This is a debug message
+    11:15:00 system.host.com SmartSim[65385] ERROR This is an error message
+    11:15:00 system.host.com SmartSim[65385] WARNING This is a warning message
+
+Notice that all log messages print to stdout. By using the highest verbosity level (`developer`),
+we instruct SmartSim to print all log messages. Remember that `developer` is extremely verbose
+logging and is intended for use during development.
```

Comment:
Are developer and debug actually the same level? If not, we should create a new message level as well so that there is a distinction. If developer and debug are the same thing, then just mention that here and only show the one example

Reply:
They are the same level, I will create a ticket

Reply:
I misread, no ticket needed but making an edit to the SS log to mention that they are the same thing